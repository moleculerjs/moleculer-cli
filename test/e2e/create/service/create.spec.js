import { describe, it, expect, beforeAll } from "vitest";
import { execa } from "execa";

const binPath = "./bin/moleculer.js";

const rm = require("rimraf");
const path = require("path");
const fs = require("fs");

const tmp = "./tmp";

describe("E2E: create command", () => {
	beforeAll(() => {
		if (!fs.existsSync(tmp)) {
			fs.mkdirSync(tmp, { mode: 0o777 });
		} else {
			rm.sync(tmp);
			fs.mkdirSync(tmp, { mode: 0o777 });
		}
	});

	it("should create js service", async () => {
		const { stderr, exitCode } =
			await execa`node ${binPath} create service --answers ${path.resolve(__dirname, "create_answers.json")}`;

		expect(exitCode).toBe(0);
		expect(stderr).toBe("");

		const svcFile = path.join(tmp, "test.service.js");

		expect(fs.existsSync(svcFile)).toBeTruthy();
		expect(fs.readFileSync(svcFile, "utf8")).toMatchSnapshot();
		// fs.unlinkSync(svcFile);
	});

	it("should create ts service", async () => {
		const { stderr, exitCode } =
			await execa`node ${binPath} create service --typescript --answers ${path.resolve(__dirname, "create_answers.json")}`;

		expect(exitCode).toBe(0);
		expect(stderr).toBe("");

		const svcFile = path.join(tmp, "test.service.ts");

		expect(fs.existsSync(svcFile)).toBeTruthy();
		expect(fs.readFileSync(svcFile, "utf8")).toMatchSnapshot();
		// fs.unlinkSync(svcFile);
	});
});
