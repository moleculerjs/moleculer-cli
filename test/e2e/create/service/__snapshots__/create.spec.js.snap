// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`E2E: create command > should create js service 1`] = `
""use strict";

/**
 * test service
 */
module.exports = {
	name: "test",

	/**
	 * Service settings
	 */
	settings: {},

	/**
	 * Service metadata
	 */
	metadata: {},

	/**
	 * Service dependencies
	 */
	dependencies: [],

	/**
	 * Actions
	 */
	actions: {
		/**
		 * Test action
		 */
		test: {
			async handler(ctx) {
				return "Hello Moleculer";
			}
		}
	},

	/**
	 * Events
	 */
	events: {
		async "some.thing"(ctx) {
			this.logger.info("Something happened", ctx.params);
		}
	},

	/**
	 * Methods
	 */
	methods: {},

	/**
	 * Service created lifecycle event handler
	 */
	created() {},

	/**
	 * Service started lifecycle event handler
	 */
	async started() {},

	/**
	 * Service stopped lifecycle event handler
	 */
	async stopped() {}
};
"
`;

exports[`E2E: create command > should create ts service 1`] = `
""use strict";

import { Service, ServiceBroker, Context } from "moleculer";

export default class TestService extends Service {
	public constructor(public broker: ServiceBroker) {
		super(broker);
		this.parseServiceSchema({
			name: "test",

			/**
			 * Service settings
			 */
			settings: {},

			/**
			 * Service metadata
			 */
			metadata: {},

			/**
			 * Service dependencies
			 */
			dependencies: [],

			/**
			 * Actions
			 */

			actions: {
				/**
				 * Test action
				 */
				test: {
					async handler(
						ctx: Context<{ name: string }>
					): Promise<string> {
						return this.ActionWelcome(ctx.params.name);
					}
				}
			}
		});
	}

	public ActionWelcome(name: string): string {
		return \`Welcome, \${name}\`;
	}
}
"
`;

exports[`E2E: start command > should create js service 1`] = `
""use strict";

/**
 * test service
 */
module.exports = {
	name: "test",

	/**
	 * Service settings
	 */
	settings: {},

	/**
	 * Service metadata
	 */
	metadata: {},

	/**
	 * Service dependencies
	 */
	dependencies: [],

	/**
	 * Actions
	 */
	actions: {
		/**
		 * Test action
		 */
		test: {
			async handler(ctx) {
				return "Hello Moleculer";
			}
		}
	},

	/**
	 * Events
	 */
	events: {
		async "some.thing"(ctx) {
			this.logger.info("Something happened", ctx.params);
		}
	},

	/**
	 * Methods
	 */
	methods: {},

	/**
	 * Service created lifecycle event handler
	 */
	created() {},

	/**
	 * Service started lifecycle event handler
	 */
	async started() {},

	/**
	 * Service stopped lifecycle event handler
	 */
	async stopped() {}
};
"
`;

exports[`E2E: start command > should create ts service 1`] = `
""use strict";

import { Service, ServiceBroker, Context } from "moleculer";

export default class TestService extends Service {
	public constructor(public broker: ServiceBroker) {
		super(broker);
		this.parseServiceSchema({
			name: "test",

			/**
			 * Service settings
			 */
			settings: {},

			/**
			 * Service metadata
			 */
			metadata: {},

			/**
			 * Service dependencies
			 */
			dependencies: [],

			/**
			 * Actions
			 */

			actions: {
				/**
				 * Test action
				 */
				test: {
					async handler(
						ctx: Context<{ name: string }>
					): Promise<string> {
						return this.ActionWelcome(ctx.params.name);
					}
				}
			}
		});
	}

	public ActionWelcome(name: string): string {
		return \`Welcome, \${name}\`;
	}
}
"
`;

exports[`test create > create js service 1`] = `
""use strict";

/**
 * test service
 */
module.exports = {
	name: "test",

	/**
	 * Service settings
	 */
	settings: {},

	/**
	 * Service metadata
	 */
	metadata: {},

	/**
	 * Service dependencies
	 */
	dependencies: [],

	/**
	 * Actions
	 */
	actions: {
		/**
		 * Test action
		 */
		test: {
			async handler(ctx) {
				return "Hello Moleculer";
			}
		}
	},

	/**
	 * Events
	 */
	events: {
		async "some.thing"(ctx) {
			this.logger.info("Something happened", ctx.params);
		}
	},

	/**
	 * Methods
	 */
	methods: {},

	/**
	 * Service created lifecycle event handler
	 */
	created() {},

	/**
	 * Service started lifecycle event handler
	 */
	async started() {},

	/**
	 * Service stopped lifecycle event handler
	 */
	async stopped() {}
};
"
`;

exports[`test create > create ts service 1`] = `
""use strict";

import { Service, ServiceBroker, Context } from "moleculer";

export default class TestService extends Service {
	public constructor(public broker: ServiceBroker) {
		super(broker);
		this.parseServiceSchema({
			name: "test",

			/**
			 * Service settings
			 */
			settings: {},

			/**
			 * Service metadata
			 */
			metadata: {},

			/**
			 * Service dependencies
			 */
			dependencies: [],

			/**
			 * Actions
			 */

			actions: {
				/**
				 * Test action
				 */
				test: {
					async handler(
						ctx: Context<{ name: string }>
					): Promise<string> {
						return this.ActionWelcome(ctx.params.name);
					}
				}
			}
		});
	}

	public ActionWelcome(name: string): string {
		return \`Welcome, \${name}\`;
	}
}
"
`;
